src/pint/scripts/event_optimize.py

```diff
@@ -241,15 +241,18 @@ def get_fit_keyvals(model, phs=0.0, phserr=0.1):
     fitkeys = [p for p in model.params if not getattr(model, p).frozen]
     fitvals = []
     fiterrs = []
-    for p in fitkeys:
+    fitunits = []
+    for i, p in enumerate(fitkeys):
         fitvals.append(getattr(model, p).value)
-        fiterrs.append(getattr(model, p).uncertainty_value)
+        fitunits.append(getattr(model, p).units)
+        fiterrs.append(float(((getattr(model, p).uncertainty).to(fitunits[i])).value))
     # The last entry in each of the fit lists is our absolute PHASE term
     # Hopefully this will become a full PINT model param soon.
     fitkeys.append("PHASE")
     fitvals.append(phs)
     fiterrs.append(phserr)
-    return fitkeys, np.asarray(fitvals), np.asarray(fiterrs)
+    fitunits.append("pulse phase")
```

Comment:
Does the string `"pulse phase"` work as an acceptable unit?

Reply:
I had originally used that as a place holder and am working on changing that as it is not an acceptable unit. Thank you for pointing that out!